indexer/src/main/java/main/Indexer.java

```diff
@@ -0,0 +1,87 @@
+package main;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import juice.modules.IndexerModule;
+import juice.modules.StrictExplicitBindingModule;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+
+import javax.inject.Singleton;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+
+@Singleton
+public class Indexer {
+
+        private final KafkaConsumer<String, String> consumer;
+        private final RestHighLevelClient client;
+
+        @Inject
+        public Indexer(KafkaConsumer<String, String> consumer, RestHighLevelClient client) {
+
+                this.consumer = consumer;
+                this.client = client;
```

Comment:
Let's do null-check for all injected members in all classes:
```suggestion
                this.consumer = requireNonNull(consumer);
                this.client = requireNonNull(client);
```

Reply:
done